src/ext/ddtrace.c

```diff
@@ -0,0 +1,169 @@
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <Zend/zend.h>
+#include <Zend/zend_exceptions.h>
+#include <php.h>
+#include <php/ext/spl/spl_exceptions.h>
+#include <php_ini.h>
+#include <ext/standard/info.h>
+
+#include "compat_zend_string.h"
+#include "ddtrace.h"
+#include "dispatch.h"
+#include "dispatch_compat.h"
+
+#include "debug.h"
+
+#define UNUSED_1(x) (void)(x)
+#define UNUSED_2(x, y) \
+    do {               \
+        UNUSED_1(x);   \
+        UNUSED_1(y);   \
+    } while (0)
+#define UNUSED_3(x, y, z) \
+    do {                  \
+        UNUSED_1(x);      \
+        UNUSED_1(y);      \
+        UNUSED_1(z);      \
+    } while (0)
+#define _GET_UNUSED_MACRO_OF_ARITY(_1, _2, _3, ARITY, ...) UNUSED_##ARITY
+#define UNUSED(...) _GET_UNUSED_MACRO_OF_ARITY(__VA_ARGS__, 3, 2, 1)(__VA_ARGS__)
+
+#if PHP_VERSION_ID < 70000
+#define PHP5_UNUSED(...) UNUSED(__VA_ARGS__)
+#else
+#define PHP5_UNUSED(...) /* unused unused */
+#endif
+
+ZEND_DECLARE_MODULE_GLOBALS(ddtrace)
+
+#define ddtrace_disabled_guard()                                                               \
+    do {                                                                                       \
+        if (DDTRACE_G(disable)) {                                                              \
+            zend_throw_exception_ex(spl_ce_RuntimeException, 0 TSRMLS_CC,                      \
+                                    "ddtrace is disabled by configuration (ddtrace.disable)"); \
+            return;                                                                            \
+        }                                                                                      \
+    } while (0)
+
+PHP_INI_BEGIN()
+STD_PHP_INI_ENTRY("ddtrace.disable", "0", PHP_INI_SYSTEM, OnUpdateBool, disable, zend_ddtrace_globals, ddtrace_globals)
```

Comment:
I assume this is around as a quick way of making all calls a noop? Is it better than just removing the extension, since this is all it does?

Reply:
Hm, this is a pattern in quite a few extensions I've seen that I followed without a second thought. 
Many PHP extensions can be compiled/statically linked into PHP itself then you need this switch to be able to enable/disable it. 

I think this extension should be able to be compiled into PHP as is, I remember adding some ifdefs to actually support it being a module.

Reply:
Ah, that makes sense I guess. No problem leaving it.